### PR TITLE
Adding testing strategies to Testing Guide.

### LINKED ISSUE #3690: [Architecture] Add documentation about Magento's approach to testing
# New topic request

## Description

Publish topic from architects regarding Magento's approach to testing.
 
## Content checklist

- [x] Topic contains information about Magento's approach to testing
- [ ] Find an appropriate place for this topic in the docs

## Additional information/resources

Content for the topic can be found here: https://github.com/magento/architecture/pull/65

### PR DESCRIPTION
**Pull request by dobooth**
## This PR is a:

- Content update

## Summary

When this pull request is merged, it will...

Add info on testing strategies to the intro Testing topic. Info taken from Architecture group who said it should be in DevDocs.
Closes #3690 

whatsnew
Added information about testing strategies to the [Magento Testing Guide](https://devdocs.magento.com/guides/v2.3/test/testing.html).

### CODE REVIEW

File: guides/v2.2/test/testing.md [hunk 1]
```diff
@@ -4,52 +4,99 @@ title: Magento Testing Guide
...
 
 Much of the functionality in Magento 2 is provided with the help of sophisticated {% glossarytooltip 312b4baf-15f7-4968-944e-c814d53de218 %}JavaScript{% endglossarytooltip %}. JavaScript tests ensure the {% glossarytooltip b00459e5-a793-44dd-98d5-852ab33fc344 %}frontend{% endglossarytooltip %} portion of Magento functions as expected.  
 
-For more information, please see the [Extension Developer Guide on JavaScript Tests]({{ page.baseurl }}/test/js/test_js-unit.html){:target="_blank"}.
+For more information, please see the [Extension Developer Guide on JavaScript Tests][].
   
 ## Static
 
-Static code analysis checks that PHP code follows the Magento 2 coding standards and best practices. They usually are executed during continuous integration using the `bin/magento` tool. 
+Static code analysis checks that PHP code follows the Magento 2 coding standards and best practices. They usually are executed during continuous integration using the `bin/magento` tool.
```
Comment (jcalcaben) [inline guides/v2.2/test/testing.md]:
```suggestion
Static code analysis checks that PHP code follows the Magento 2 coding standards and best practices.
They are executed during continuous integration using the `bin/magento` tool.
```
Split sentences into separate line and remove word to strengthen sentence.

File: guides/v2.2/test/testing.md [hunk 2]
```diff
@@ -4,52 +4,99 @@ title: Magento Testing Guide
...
 
-Please see the [`magento dev:tests:run`]({{ page.baseurl }}/config-guide/cli/config-cli-subcommands-test.html){:target="_blank"} documentation for more information, using the test type `static`.
+Please see the [`magento dev:tests:run`][] documentation for more information, using the test type `static`.
 
 ## Unit
 
-Unit tests are used to check a single unit of PHP code in isolation. They are usually written during development using [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development){:target="_blank"} (TDD).  
+Unit tests are used to check a single unit of PHP code in isolation. They are usually written during development using [test-driven development][] (TDD).
```
Comment (jcalcaben) [inline guides/v2.2/test/testing.md]:
```suggestion
Unit tests are used to check a single unit of PHP code in isolation.
They are written during development using [test-driven development][] (TDD).  
```

Split sentences into separate lines.

File: guides/v2.2/test/testing.md [hunk 3]
```diff
@@ -4,52 +4,99 @@ title: Magento Testing Guide
...
 Because they do not require the full Magento application stack to be initialized, they run an order of magnitude faster then integration tests.  
 
-For more information, see [Running Unit Tests]({{ page.baseurl }}/test/unit/unit_test_execution.html){:target="_blank"}.
+For more information, see [Running Unit Tests][].
 
-Please refer to the article [Writing testable code]({{ page.baseurl }}/test/unit/writing_testable_code.html){:target="_blank"} for more information on what to keep in mind when starting with TDD.
+Please refer to the article [Writing testable code][] for more information on what to keep in mind when starting with TDD.
   
-The `bin/magento` tool provides a common entry point to execute any of the tests, which can be useful for continuous integration. Please see the [System Administrators Guide on Running Tests]({{ page.baseurl }}/config-guide/cli/config-cli-subcommands-test.html){:target="_blank"} for more information.
+The `bin/magento` tool provides a common entry point to execute any of the tests, which can be useful for continuous integration. Please see the [System Administrators Guide on Running Tests][] for more information.
```
Comment (jcalcaben) [inline guides/v2.2/test/testing.md]:
```suggestion
The `bin/magento` tool provides a common entry point to execute any of the tests, which can be useful for continuous integration.
Please see the [System Administrators Guide on Running Tests][] for more information.
```
Split sentences into separate lines.

File: guides/v2.2/test/testing.md (hunk 2 above)
Comment (dobooth) [reply guides/v2.2/test/testing.md]: Thanks.

**Comment by dobooth** [comment]
running tests

**Pull request merged**